src/util/RouterUtil.js

```diff
@@ -133,12 +133,22 @@ function (Okta, OAuth2Util, Util, Enums, BrowserFeatures, Errors, ErrorCodes) {
 
       var redirectFn = router.settings.get('redirectUtilFn');
 
-      var nextUrl = res._links && res._links.next && res._links.next.href;
-      if (nextUrl) {
+      var nextUrl = res._links && ((res._links.original && res._links.original.href) || (res._links.next && res._links.next.href));
+      if (res.type === Enums.SESSION_STEP_UP) {
+        var targetUrl = res._links && res._links.next && res._links.next.href;
+        successData.stepUp = {
+          url: targetUrl,
+          finish: function () {
+            redirectFn(targetUrl);
+          }
+        }
+      } else if (nextUrl) {
         successData.next = function () {
           redirectFn(nextUrl);
         };
       } else {
+        // Add the type for now until the API returns it.
+        successData.type = Enums.SESSION_SSO;
```

Comment:
@haishengwu-okta Added this back here, so will not be breaking change.
I would also keep the `||` condition in line https://github.com/okta/okta-signin-widget/pull/619/commits/3f7564633e4a8124290eeca6ac8211f4522c19e7#diff-1d1f2f1a166e47fd617d3b5e9dfe4b74R127 so that the in new if case (original or next link) will behave in that new way, since we will have new response.type in future, and we probably don't want to force `Enums.SESSION_SSO` everytime, what do you think?

Reply:
I'm kind of hesitate and prefer revisit the code when the future comes.

Reply:
Just as an FYI, I think the true ideal situation is that the widget does not care what the `type` is at all, as it should not need to behave differently.